src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -132,11 +116,42 @@ public Test BuildFrom(FixtureMethod method, Test suite)
                     tests.Add(test); 
             }
 
-            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || tests.Count > 0
-                ? BuildParameterizedMethodSuite(method, tests)
+            return builders.Count > 0 && method.Method.GetParameters().Length > 0 || tests.Count > 0 
+                ? BuildParameterizedMethodSuite(method, tests) 
                 : BuildSingleTestMethod(method, suite);
         }
 
+        /// <summary>
+        /// Checks to see if the method has arguments with DataAttribute, 
+        /// if the argument has no data, the test should not run rather than fail.
+        /// </summary>
+        /// <param name="fixtureMethod">The method for which a test is to be built.</param>
+        /// <returns>True if the method signature contains DataAttribute arguments</returns>
+        private bool MethodHasDataAttributeParameters(FixtureMethod fixtureMethod)
```

Comment:
Can we change the method name and comments to reflect that we are looking for `IParameterDataSource` and not `DataAttribute`?